akka-projection-cassandra/src/main/scala/akka/projection/cassandra/internal/CassandraProjectionImpl.scala

```diff
@@ -73,11 +76,12 @@ import akka.stream.scaladsl.Source
     strategy: CassandraProjectionImpl.Strategy,
     settingsOpt: Option[ProjectionSettings],
     handler: Handler[Envelope])
-    extends Projection[Envelope] {
+    extends javadsl.CassandraProjection[Envelope]
+    with scaladsl.CassandraProjection[Envelope] {
   import CassandraProjectionImpl._
   import HandlerRecoveryImpl.applyUserRecovery
 
-  override def withSettings(settings: ProjectionSettings): Projection[Envelope] = {
+  override def withSettings(settings: ProjectionSettings): CassandraProjectionImpl[Offset, Envelope] = {
```

Comment:
```suggestion
  override def withSettings(settings: ProjectionSettings): CassandraProjection[Offset, Envelope] = {
```

Reply:
that's a trick for javadsl/scaladsl
withSettings is overridden in the javadsl.CassandraProjection and scaladsl.CassandraProjection, and that is what the end user will see

Reply:
Ok, I see. Not a big issue.